nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/hadoop/SecurityUtil.java

```diff
@@ -141,4 +143,37 @@ public static boolean isSecurityEnabled(final Configuration config) {
         Validate.notNull(config);
         return KERBEROS.equalsIgnoreCase(config.get(HADOOP_SECURITY_AUTHENTICATION));
     }
+
+    public static <T> T callWithUgi(CallableThrowingException<UserGroupInformation, IOException> ugiProvider, CallableThrowingException<T, IOException> function) throws IOException {
+        try {
+            return ugiProvider.call().doAs((PrivilegedExceptionAction<T>)() -> function.call());
```

Comment:
minor nit: `() -> function.call()` could be replaced with method reference, i.e. `function::call`